src/utils/dolphin.ts

```diff
@@ -22,6 +22,8 @@ interface DolphinCombo {
   endFrame?: number;
   gameStation?: string;
   gameStartAt?: string;
+  damageDone?: number;
+  comboingCharacter?: number;
```

Comment:
I'm a bit confused about these values. `gameStation` and `gameStartAt` are values actually used by Dolphin, right? Something about the `outputOverlayFiles` option. On the other hand `damageDone` and `comboingCharacter` aren't used by Dolphin right? But are only used to filter down the existing combos in the queue? Is my understanding correct?

I think we should distinguish between the values that Dolphin accepts (which are the values we will right into the JSON file) and the values that we use internally but don't write out to the JSON file.

Reply:
What I think would be a good solution, is to track a list combos, storing the file path, the combo object itself (which includes damage done and character info), as well as any relevant metadata, that's what we store internally in the class.

Then we can apply filters all we want to on the original combo objects.

Only when we call the `writeFile()` function, that's when we map those raw combo objects, into the format accepted by Dolphin, we change the start/end frames etc only at this stage, and not at the `addCombo` stage.

Reply:
you're correct that dolphin doesn't use them, I'm mainly just trying to bring up features from getSlippiCombos that were added after you made this library.

Your solution sounds like it would be best. I can maybe implement it if you don't have time, but I might need some guidance here and there.